syntax = "proto3";

message File{
  string file_name = 1;
}

message UserData{
  string username = 1;
  string password = 2;
}

message Username{
  string username = 1;
}

message Response{
  int32 status_code = 1;
}

service PServer{
  rpc Login(UserData) returns (Response);
  rpc Logout(Username) returns (Response);

  rpc DownloadFileRequest(File) returns (Response);
  rpc DownloadFile(File) returns (Response);

  rpc UploadFileRequest(File) returns (Response);
  rpc UploadFile(File) returns (Response);
}